src/WithdrawRequestNFT.sol

```diff
@@ -93,63 +127,48 @@ contract WithdrawRequestNFT is ERC721Upgradeable, UUPSUpgradeable, OwnableUpgrad
         IWithdrawRequestNFT.WithdrawRequest memory request = _requests[tokenId];
         require(request.isValid, "Request is not valid");
 
-        uint256 fee = uint256(request.feeGwei) * 1 gwei;
         uint256 amountToWithdraw = getClaimableAmount(tokenId);
 
         // transfer eth to recipient
         _burn(tokenId);
         delete _requests[tokenId];
-
-        if (fee > 0) {
-            // send fee to membership manager
-            liquidityPool.withdraw(address(membershipManager), fee);
-        }
+        
+        uint256 shareAmountToBurnForWithdrawal = liquidityPool.sharesForWithdrawalAmount(amountToWithdraw);
+        totalLockedEEthShares -= shareAmountToBurnForWithdrawal;
 
         uint256 amountBurnedShare = liquidityPool.withdraw(recipient, amountToWithdraw);
-        uint256 amountUnBurnedShare = request.shareOfEEth - amountBurnedShare;
-        if (amountUnBurnedShare > 0) {
-            accumulatedDustEEthShares += uint96(amountUnBurnedShare);
-        }
+        assert (amountBurnedShare == shareAmountToBurnForWithdrawal);
 
-        emit WithdrawRequestClaimed(uint32(tokenId), amountToWithdraw + fee, amountBurnedShare, recipient, fee);
+        emit WithdrawRequestClaimed(uint32(tokenId), amountToWithdraw, amountBurnedShare, recipient, 0);
     }
 
-    function batchClaimWithdraw(uint256[] calldata tokenIds) external {
+    function batchClaimWithdraw(uint256[] calldata tokenIds) external whenNotPaused {
         for (uint256 i = 0; i < tokenIds.length; i++) {
             _claimWithdraw(tokenIds[i], ownerOf(tokenIds[i]));
         }
     }
 
-    // a function to transfer accumulated shares to admin
-    function burnAccumulatedDustEEthShares() external onlyAdmin {
-        require(eETH.totalShares() > accumulatedDustEEthShares, "Inappropriate burn");
-        uint256 amount = accumulatedDustEEthShares;
-        accumulatedDustEEthShares = 0;
+    // This function is used to aggregate the sum of the eEth shares of the requests that have not been claimed yet.
+    // To be triggered during the upgrade to the new version of the contract.
+    function aggregateSumEEthShareAmount(uint256 _numReqsToScan) external {
```

Comment:
Remainder shares will still accumulate in the the contract after this upgrade for future withdrawal requests due to the known diff between current share price and share price at time of request. Because of this, the admin will have to continuously call `aggregateSumEEthShareAmount` and `handleRemainder`.

The current implementation can only handle remainders for requests made before the upgrade as `_lastRequestIdToScanUntilForShareRemainder` is only set once in the initialization function:
```
_lastRequestIdToScanUntilForShareRemainder = nextRequestId - 1;
```